### DISCUSSION TITLE
autosuggestion for Ruby keyword "end" seems broken

### DISCUSSION THREAD

**Discussion by nperspective** [neoclide/coc.nvim#4707] [Q&A] [answered]
Simple issue, maybe has been addressed already, but I did a quick search and could not find reference to this, as maybe it's expected functionality. 

Issue: when I type the Ruby keyword 'end', the autosuggestion window will highlight the top result which is some random method or variable from my code above, such as 'evenly_div_by_3' (seen below) and, when I hit enter/return, my typed 'end' word is overwritten with the top autosuggestion result, unless I use the escape key to exit the autosuggestions window.

![Screenshot 2023-07-19 at 2 11 32 PM](https://github.com/neoclide/coc.nvim/assets/49545456/148b0ed8-0fca-47c9-9218-5582b938f3c5)

Expected / Desired behavior: when I type 'end', the autosuggestion window will highlight exactly what I typed: 'end' so when I hit enter/return, 'end' is left in the editor....or it detects the Ruby 'end' keyword and simply does not display the autosuggestions window at all.

What setting / config do I need to change for this to behave?

**Comment by fannheyward** [comment, marked as answer]
coc.nvim is a LSP client that needs language server as completion sources to do completions. You need to setup a Ruby language server, checkout servers: https://github.com/neoclide/coc.nvim/wiki/Language-servers#ruby

**Reply by nperspective** [reply]
sick! thank you for explaining it so simply. It's working now :)
